### PR TITLE
Usability improvements for no series

### PR DESCRIPTION
**Pull request by AndreasMoth**
Adding a success/failure message to Check No. Series, promoting Lines and Relations actions on No. Series page to always show and adding a FactBox on No. Series page to show the relations a No. Series has.

Fixes [AB#476364](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/476364)

### CODE REVIEW

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al [hunk 1]
```diff
@@ -180,12 +181,18 @@ page 456 "No. Series"
...
                         Rec.TestField(Code);
                         NoSeriesSetupImpl.SetImplementation(Rec, Implementation);
+                        UpdateLineActionOnPage();
                     end;
                 }
             }
         }
         area(FactBoxes)
         {
+            part(NoSeriesRelationsFactBox; "No. Series Relationships Part")
```
Comment (AndreasMoth) [inline src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: I was wondering if it also makes sense to add a part with the lines, thoughts?

Comment (AndreasMoth) [reply src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: Could be a condensed version with just Starting No., Ending No., Last No. Used

Comment (NKarolak) [reply src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: What is the use case? The lines are already displayed on the page.
image

Comment (grobyns) [reply src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: might be nice to see the lines yes. Also, could add that same info but for 'current line' to the relationship part ... might have to do some background processing if its slow though...

Comment (grobyns) [reply src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: @NKarolak A line is displayed, based on Workdate. You likely have many lines. The part could show all open lines.

Comment (AndreasMoth) [reply src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: Adding No. Series Lines part, adding all fields it currently has but hiding everything except the three fields Starting No., Ending No., Last No. Used. This allows customers to show any fields they need for their task.

Comment (AndreasMoth) [reply src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: The page now looks like this:
image

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al [hunk 2]
```diff
@@ -282,6 +307,8 @@ page 456 "No. Series"
         LastDateUsed: Date;
         AllowGaps: Boolean;
         Implementation: Enum "No. Series Implementation";
+        CheckNoSucceededTxt: Label 'Checking the next no. succeeded and returned %1 for WorkDate %2', Comment = '%1 = A No. Series number, %2 = a date';
```
Comment (darjoo) [inline src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: `Checking the next no. succeeded and returned %1 for WorkDate %2`
Change to 
`Checking the next no. succeeded and returned %1 for work date %2`

Double check with Jacob and team for text. WorkDate is weird to the end users. "work date" is more normal.

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al [hunk 3]
```diff
@@ -282,6 +307,8 @@ page 456 "No. Series"
...
+        CheckNoFailedTxt: Label 'Checking the next no. failed for WorkDate %1', Comment = '%1 = a date';
```
Comment (darjoo) [inline src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: Same with this text.

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al (hunk 2 above)
Comment (AndreasMoth) [reply src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: Double checked with Brent and got a nice label.

**Review by darjoo — DISMISSED** [review]

**Review by grobyns — APPROVED** [review]

**Review by grobyns — DISMISSED** [review]

**Review by JesperSchulz — APPROVED** [review]

**Review by darjoo — APPROVED** [review]

File: src/Business Foundation/Test/NoSeries/src/ERMNoSeriesTests.Codeunit.al [hunk 4]
```diff
@@ -1074,10 +1076,15 @@ codeunit 134370 "ERM No. Series Tests"
...
     local procedure Initialize()
     begin
         PermissionsMock.Set('No. Series - Admin');
         DeleteNumberSeries('TEST');
     end;
+
+    [MessageHandler]
+    procedure TestSeriesSuccessMessageHandler(Message: Text[1024])
+    begin
+        LibraryAssert.IsTrue(Message.StartsWith('The test was successful.'), 'The test series was not successful, message: ' + Message);
```
Comment (grobyns) [inline src/Business Foundation/Test/NoSeries/src/ERMNoSeriesTests.Codeunit.al:1088]:
```suggestion
        LibraryAssert.ExpectedMessage('The test was successful.', Message);
```

**Review by grobyns — CHANGES_REQUESTED** [review]

File: src/Business Foundation/Test/NoSeries/src/ERMNoSeriesTests.Codeunit.al (hunk 4 above)
Comment (grobyns) [reply src/Business Foundation/Test/NoSeries/src/ERMNoSeriesTests.Codeunit.al:1088]: Also, just copy the label and use that...

**Pull request merged**